core/extension/extension_api_dump.cpp

```diff
@@ -1056,4 +1065,275 @@ void GDExtensionAPIDump::generate_extension_json_file(const String &p_path) {
+	if (!p_old_api.has(p_outer)) {
+		return true; // May just not have this array and its still good. Probably added recently or optional.
+	}
+	bool failed = false;
+	ERR_FAIL_COND_V_MSG(!p_new_api.has(p_outer), false, "New API lacks base array: " + p_outer);
+	Array new_api = p_new_api[p_outer];
+	HashMap<String, Dictionary> new_api_assoc;
+
+	for (int i = 0; i < new_api.size(); i++) {
+		Dictionary elem = new_api[i];
+		ERR_FAIL_COND_V_MSG(!elem.has(p_outer_name), false, "Validate extension JSON: Element of base_array '" + p_outer + "' is missing field '" + p_outer_name + "'. This is a bug.");
+		new_api_assoc.insert(elem[p_outer_name], elem);
+	}
+
+	Array old_api = p_old_api[p_outer];
+
+	for (int i = 0; i < old_api.size(); i++) {
+		Dictionary old_elem = old_api[i];
+		if (!old_elem.has(p_outer_name)) {
+			failed = true;
+			print_error("Validate extension JSON: JSON file: element of base array '" + p_outer + "' is missing the field: '" + p_outer_name + "'.");
+			continue;
+		}
+		String name = old_elem[p_outer_name];
+		if (!new_api_assoc.has(name)) {
+			failed = true;
+			if (!r_removed_classes_registered.has(name)) {
+				print_error("Validate extension JSON: API was removed: " + p_outer + "/" + name);
+				r_removed_classes_registered.insert(name);
+			}
+			continue;
+		}
+
+		Dictionary new_elem = new_api_assoc[name];
+
+		if (!compare_dict_array(old_elem, new_elem, p_base_array, p_name_field, p_fields_to_compare, p_compare_hashes, p_outer + "/" + name + "/", p_compare_operators)) {
+			failed = true;
+		}
+	}
+
+	return !failed;
+}
+
+Error GDExtensionAPIDump::validate_extension_json_file(const String &p_path) {
+	Error error;
+	String text = FileAccess::get_file_as_string(p_path, &error);
+	if (error != OK) {
+		ERR_PRINT("Validate extension JSON: Could not open file '" + p_path + "'.");
+		return error;
+	}
+
+	Ref<JSON> json;
+	json.instantiate();
+	error = json->parse(text);
+	if (error != OK) {
+		ERR_PRINT("Validate extension JSON: Error parsing '" + p_path + "' at line " + itos(json->get_error_line()) + ": " + json->get_error_message());
+		return error;
+	}
+
+	Dictionary old_api = json->get_data();
+	Dictionary new_api = generate_extension_api();
+
+	{ // Validate header:
+		Dictionary header = old_api["header"];
+		ERR_FAIL_COND_V(!header.has("version_major"), ERR_INVALID_DATA);
+		ERR_FAIL_COND_V(!header.has("version_minor"), ERR_INVALID_DATA);
+		int major = header["version_major"];
+		int minor = header["version_minor"];
+
+		ERR_FAIL_COND_V_MSG(major != VERSION_MAJOR, ERR_INVALID_DATA, "JSON API dump is for a different engine version (" + itos(major) + ") than this one (" + itos(major) + ")");
+		ERR_FAIL_COND_V_MSG(minor >= VERSION_MINOR, ERR_INVALID_DATA, "JSON API dump is for a newer version of the engine: " + itos(major) + "." + itos(minor));
```

Comment:
```suggestion
		ERR_FAIL_COND_V_MSG(minor > VERSION_MINOR, ERR_INVALID_DATA, "JSON API dump is for a newer version of the engine: " + itos(major) + "." + itos(minor));
```

or you can't check compatibility between patch versions